src/decode.rs

```diff
@@ -4388,8 +4377,8 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
                 }
                 f.frame_thread.pal_idx_sz = pal_idx_sz;
             }
-        } else if !f.frame_thread.pal.is_null() {
-            rav1d_freep_aligned(&mut f.frame_thread.pal as *mut *mut [[u16; 8]; 3] as *mut c_void);
+        } else if !f.frame_thread.pal.is_empty() {
+            let _ = mem::take(&mut f.frame_thread.pal);
             rav1d_freep_aligned(&mut f.frame_thread.pal_idx as *mut *mut u8 as *mut c_void);
             f.frame_thread.pal_idx_sz = 0;
             f.frame_thread.pal_sz = f.frame_thread.pal_idx_sz;
```

Comment:
`.pal_sz` can be replaced with a method,

```rust
impl Rav1dFrameContext_frame_thread {
    pub fn pal_sz(&self) -> usize {
        self.pal.len() / (16 * 16)
    }
}
```

Reply:
Done!